here-naksha-app-service/src/test/java/com/here/naksha/app/service/ViewHandlerTest.java

```diff
@@ -0,0 +1,167 @@
+        //given
+        final String idsQueryParam = "id=my-custom-id-5003-1";
+        final String streamId = UUID.randomUUID().toString();
+        final String expectedBodyPart = loadFileOrFail("ViewHandler/TC5003_searchById/feature_response_part.json");
+
+        //when
+        HttpResponse<String> viewResponse = getNakshaClient().get("hub/spaces/" + SPACE_ID + "/features?" + idsQueryParam, streamId);
+
+        //then
+        assertThat(viewResponse)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedBodyPart, "Create Feature response body doesn't match");
+    }
+    @Test
+    void tc5004_searchById_GetFromDlbWhenAvailableAtDlbAndBase() throws Exception {
+        //given
+        final String idsQueryParam = "id=my-custom-id-5004-1";
+        final String streamId = UUID.randomUUID().toString();
+        final String expectedBodyPart = loadFileOrFail("ViewHandler/TC5004_searchById/feature_response_part.json");
+
+        //when
+        HttpResponse<String> viewResponse = getNakshaClient().get("hub/spaces/" + SPACE_ID + "/features?" + idsQueryParam, streamId);
+
+        //then
+        assertThat(viewResponse)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedBodyPart, "Create Feature response body doesn't match");
+    }
+
+    @Test
+    void tc5005_searchById_GetFromBaseWhenAvailableOnlyAtBase() throws Exception {
+        //given
+        final String idsQueryParam = "id=my-custom-id-5005-1";
+        final String streamId = UUID.randomUUID().toString();
+        final String expectedBodyPart = loadFileOrFail("ViewHandler/TC5005_searchById/feature_response_part.json");
+
+        //when
+        HttpResponse<String> viewResponse = getNakshaClient().get("hub/spaces/" + SPACE_ID + "/features?" + idsQueryParam, streamId);
+
+        //then
+        assertThat(viewResponse)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedBodyPart, "Create Feature response body doesn't match");
+    }
+
+    @Test
+    void tc5006_searchById_ResultFromMultipleSpaces() throws Exception {
+        //given
+        final String idsQueryParam = "id=my-custom-id-5002-1&id=my-custom-id-5004-1&id=my-custom-id-5005-1";
+        final String streamId = UUID.randomUUID().toString();
+        final String expectedBodyPart = loadFileOrFail("ViewHandler/TC5006_searchById/feature_response_part.json");
+
+        //when
+        HttpResponse<String> viewResponse = getNakshaClient().get("hub/spaces/" + SPACE_ID + "/features?" + idsQueryParam, streamId);
+
+        //then
+        assertThat(viewResponse)
+                .hasStatus(200)
+                .hasStreamIdHeader(streamId)
+                .hasJsonBody(expectedBodyPart, "Create Feature response body doesn't match");
+    }
+}
```

Comment:
As per Jira scope, we need BBox related test as well, where we ensure that the features are retrieved as per layered priority (Delta, DLB and Consistent store).

Reply:
Tests for bbox provided